### ISSUE TITLE
sql code cells feature

### ISSUE THREAD

**Issue by mahboiterare** [quarto-dev/quarto-vscode#75]
Hello,
I am trying to reproduce some of my reports and one feature I fail to spot is the ability to create SQL code cells like in RMarkdown. 
I thought that if quarto used knitr to produce views (and knitr supports creation of sql cells), then sql chunk would be an option, yet I struggle to recreate this kind of behavior. Is there a possibility to create sql scripts in this fashion?

```{sql}
{sql}
#| connection = con
SELECT 1 FROM test
```

Thanks!

**Comment by jjallaire** [comment]
You might need to use the inline cell syntax as that supports R code/objects:

````
```{sql, connection = con}
SELECT 1 FROM test
```
````

**Comment by jjallaire** [comment]
@cderv and @yihui I'm wondering if there are attributes like `connection` that can _only_ be an R expression whether we should have those pre-registered and evaluated as such. This is sort of like for htmlwidgets where if we know a property can only be JavaScript we automatically eval it (e.g. something that takes a function as its value)

**Comment by mahboiterare** [comment]
Okay, I tried it out in RStudio IDE and it works there, however, the problem is that vs code does not allow to run those sql chunks; I will try to dig into vs code settings
![image](https://user-images.githubusercontent.com/62305232/188813670-73472245-a912-406d-a5c1-9c2b79508588.png)

**Comment by jjallaire** [comment]
VS Code indeed does not know how to run SQL chunks.

**Comment by mahboiterare** [comment]
Well, I am trying to dig deeper and quarto extension even has preloaded code snippet for sql chunks and it is strange for me that there is no way one can execute it.
![image](https://user-images.githubusercontent.com/62305232/188912068-6d37e5a3-c75e-47e6-b5b0-edbc32188b83.png)

![image](https://user-images.githubusercontent.com/62305232/188911971-61b5d31c-7e8a-4305-b8e3-3987913c0605.png)

Sorry, if I am asking stupid questions, but for me sql support is a killer-feature as I could be using vscode as all-in-one editor for my data analysis

**Comment by jjallaire** [comment]
Yes, agree this is a killer feature and one we would like to build (but it's not a small feature so might be a while)

**Comment by yihui** [comment]
> @cderv and @yihui I'm wondering if there are attributes like `connection` that can _only_ be an R expression whether we should have those pre-registered and evaluated as such.

`connection` can be either an R object or a character string (object name): https://github.com/yihui/knitr/blob/6bfffe9c1ae8c84f0f2ae07cd3c4b00876757587/R/engine.R#L601-L602

so any of the following syntax variants should work:

```{sql}
#| connection = con
SELECT 1 FROM test
```

```{sql}
#| connection = "con"
SELECT 1 FROM test
```

```{sql}
#| connection: !expr con
SELECT 1 FROM test
```

```{sql}
#| connection: "con"
SELECT 1 FROM test
```

```{sql, connection = con}
SELECT 1 FROM test
```

```{sql, connection = "con"}
SELECT 1 FROM test
```

**Comment by jjallaire** [comment]
Okay, very good to know!!!

**Comment by cderv** [comment]
Related issue in Quarto : https://github.com/quarto-dev/quarto-cli/issues/2310

Not all the form above will work in Quarto currently. 

* Passing unquoted variable will throw an error ( reported in https://github.com/quarto-dev/quarto-cli/issues/2310) 
* Using in-chunk YAML for `sql` engine would work when quoted, but RStudio IDE does not handle correctly the YAML section interactively (https://github.com/rstudio/rstudio/issues/11606)

So for now the only form seems to be working 
````
```{sql, connection = "con"}
SELECT 1 FROM test
```
````

We should improve that in Quarto CLI but also in VS Code when it will be possible, and in RStudio IDE too. 

Sharing feedback here for cross repo references
